lib/src/source_visitor.dart

```diff
@@ -108,12 +108,45 @@ class SourceVisitor extends ThrowingAstVisitor {
 
   visitAdjacentStrings(AdjacentStrings node) {
     builder.startSpan();
-    builder.startRule();
-    visitNodes(node.strings, between: splitOrNewline);
+    var rule = new Rule();
+    builder.startRule(rule);
+
+    var chunk = zeroSplit();
```

Comment:
> ```dart
> {
>   "long adjacent" "long adjacent";
> }
> ```
> 
> That should not get formatted to:
> 
> ```dart
> {
>   // <-- split added here.
>   "long adjacent"
>   "long adjacent";
> }
> ```

That's probably true, but this seems like an extremely narrow edge-case considering that such a statement can't possibly do anything.

> Likewise:
> 
> ```dart
> var x = foo ? "short" : "long adjacent" "long adjacent";
> ```
> 
> Should maybe not become:
> 
> ```dart
> var x = foo
>     ? "short"
>     :
>         "long adjacent"
>         "long adjacent";
> ```
> 
> Or maybe it should?

I'd hand-format this as

```dart
var x = foo
    ? "short"
    : "long adjacent"
      "long adjacent";
```

I could try to add a special case to do that, but I don't think the formatting you listed is too bad. 

> It may be that the best way to handle this is to have ArgumentListVisitor specifically handle arguments that are adjacent strings.

This would make a fair number of `return` and `throw` cases look worse :confused:.